File: demo-web-java/src/main/java/com/example/demowebjava/security/CustomUserDetails.java
package com.example.demowebjava.security;

import com.example.demowebjava.model.User;
import lombok.Getter;
import lombok.ToString;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.HashSet;
import java.util.Set;

@Getter
@ToString
public class CustomUserDetails implements UserDetails {
 private Long id;
 private String username;
 private String password;
 private Set<GrantedAuthority> authorities;

 public CustomUserDetails(User user) {
   this.id = user.getId();
   this.username = user.getUsername();
   this.password = user.getPassword();
   authorities = new HashSet<>();
   user.getAuthorities().stream().forEach(authority -> authorities.add((GrantedAuthority) authority::name));
 }

 @Override public boolean isAccountNonExpired() { return true; }
 @Override public boolean isAccountNonLocked() { return true; }
 @Override public boolean isCredentialsNonExpired() { return true; }
 @Override public boolean isEnabled() { return true; }
}

File: demo-web-scala/src/main/scala/com/example/demo/security/CustomUserDetails.scala
package com.example.demo.security

import com.example.demo.model.User
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

import scala.beans.BeanProperty

class CustomUserDetails(user: User) extends UserDetails{

  var id: Long = _

  @BeanProperty
  var username: String = _

  @BeanProperty
  var password: String = _

  @BeanProperty
  var authorities: java.util.Collection[GrantedAuthority] = new java.util.HashSet[GrantedAuthority]()

  id = user.getId()
  username = user.getName()
  password = user.getPassword()
  user.getAuthorities().stream().forEach(o => authorities.add(new GrantedAuthority {
    override def getAuthority: String = o.name()
  }))

  override def isAccountNonExpired = true

  override def isAccountNonLocked = true

  override def isCredentialsNonExpired = true

  override def isEnabled = true

}
